flink-core/src/main/java/org/apache/flink/api/common/operators/CollectionExecutor.java

```diff
@@ -501,4 +536,22 @@ public int getSuperstepNumber() {
 			return (T) previousAggregates.get(name);
 		}
 	}
+
+	private static final class DoingNothing implements Callable<Path>{
+		private Path entry;
+
+		public DoingNothing(Path entry){
+			this.entry = entry;
+		}
+
+		@Override
+		public Path call() throws IOException{
+			try{
+				LocalFileSystem fs = (LocalFileSystem) entry.getFileSystem();
+				return entry.isAbsolute() ? new Path(entry.toUri().getPath()): new Path(fs.getWorkingDirectory(),entry);
+			} catch (ClassCastException e){
+				throw new RuntimeException("Collection execution must have only local file paths");
```

Comment:
dislike this error message, there's is no apparent relation to the distributed cache. 
how about "The DistrbutedCache only supports local files for Collection Environments."